crates/ra_hir_ty/src/lower.rs

```diff
@@ -805,8 +806,9 @@ fn fn_sig_for_struct_constructor(db: &impl HirDatabase, def: StructId) -> PolyFn
 /// Build the type of a tuple struct constructor.
 fn type_for_struct_constructor(db: &impl HirDatabase, def: StructId) -> Binders<Ty> {
     let struct_data = db.struct_data(def.into());
-    if struct_data.variant_data.is_unit() {
-        return type_for_adt(db, def.into()); // Unit struct
+    match struct_data.variant_data.kind() {
```

Comment:
this could be an `if let` or `if ... == StructKind::Unit`

Reply:
Just changed it. Out of curiosity - what makes `if let` preferable? Just that it's more concise? I would think it a lot of situations `match` without a default is better because you'll get a compiler error if a new variant is added which forces you to explicitly decide how to handle the new variant. Of course, that doesn't really apply here because `StructKind` isn't likely to change.

Reply:
I think in this case, as you said, `StructKind` isn't likely to change, and we just want to handle the unit case as a special case here.